src/libs/signer.service.ts

```diff
@@ -34,25 +43,47 @@ export async function signTransaction(signingRequest: SigningRequest) {
   const { to, gasLimit, transactionData } = signingRequest;
 
   const signerAddress = signer.address;
+  logger.info(`🖋️ Signing transaction with signing key ${signerAddress}`);
 
-  const signerNonce = await provider.getTransactionCount(signerAddress);
+  let signerNonce: number;
+  try {
+    signerNonce = await provider.getTransactionCount(signerAddress);
+  } catch (error) {
+    logger.error("❌ Unable to get signing key nonce.");
+    throw error;
+  }
 
-  const txRequest = {
+  const transactionParameters = {
     to,
     from: signerAddress,
     nonce: signerNonce,
     gasLimit,
-    value: ethers.utils.hexlify(0),
+    value: 0,
     type: 2,
-    chainId: (await provider.getNetwork()).chainId,
+    chainId: Number.parseInt(CHAIN_ID),
```

Comment:
maybe we can just remove the CHAIN_ID value entirely and just fetch it from the provider

Reply:
In which file you would find it relevant to add this function ?